src/Symfony/Component/Notifier/Bridge/Amazon/AmazonTransportFactory.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\Amazon;
+
+use AsyncAws\Core\Configuration;
+use AsyncAws\Sns\SnsClient;
+use Symfony\Component\Notifier\Exception\UnsupportedSchemeException;
+use Symfony\Component\Notifier\Transport\AbstractTransportFactory;
+use Symfony\Component\Notifier\Transport\Dsn;
+use Symfony\Component\Notifier\Transport\TransportInterface;
+
+/**
+ * @author Adrien Chinour <github@chinour.fr>
+ *
+ * @experimental in 5.3
+ */
+final class AmazonTransportFactory extends AbstractTransportFactory
+{
+    private const DSN_SCHEME = 'sns';
+
+    public function create(Dsn $dsn): TransportInterface
+    {
+        $scheme = $dsn->getScheme();
+
+        if (self::DSN_SCHEME === $scheme) {
+            $options = [
```

Comment:
What's about something similar to `SES`?
```php
$options = [
  'region' => $dsn->getOption('region') ?: 'eu-west-1',
  'profile' => $dsn->getOption('profile'),
  'accessKeyId' => $dsn->getUser(),
  'accessKeySecret' => $dsn->getPassword(),
] + (
  'default' === $dsn->getHost() ? [] : ['endpoint' => 'https://'.$dsn->getHost().($dsn->getPort() ? ':'.$dsn->getPort() : '')]
);
```

No need to define a default value, `async-aws` use it own default value when provided value is null.